frontend/src/App.test.tsx

```diff
@@ -860,22 +861,24 @@ describe("App.handleNewSession", () => {
       instance.connectionManager.getBaseUriParts = mockGetBaseUriParts()
 
       window.history.pushState({}, "", "/")
-      jest.spyOn(
+      pushStateSpy = jest.spyOn(
         window.history,
         // @ts-ignore
         "pushState"
       )
     })
 
     afterEach(() => {
+      // @ts-ignore
+      pushStateSpy.mockRestore()
       window.history.pushState({}, "", "/")
     })
 
     it("can switch to the main page", () => {
       const instance = wrapper.instance() as App
       instance.handleNewSession(new NewSession(NEW_SESSION_JSON))
 
-      expect(window.history.pushState).toHaveBeenCalledWith({}, "", "/")
+      expect(window.history.pushState).toHaveBeenLastCalledWith({}, "", "/")
```

Comment:
Assertion on `window.history.*` should be `toHaveBeenLastCalledWith` instead because the target object is global and `toHaveBeenCalledWith` can match older function calls in other tests if we forget resetting the mock.
This can't be a problem as long as we reset the mock after each unit test properly, but a safer way should be used as fail-safe.